clang/include/clang/AST/ASTContext.h

```diff
@@ -2381,6 +2384,15 @@ class ASTContext : public RefCountedBase<ASTContext> {
     return getTagDeclType(MSGuidTagDecl);
   }
 
+  /// Retrieve the implicitly-predeclared 'struct type_info' declaration.
+  TagDecl *getMSTypeInfoTagDecl() const {
+    // Lazily create this type on demand - it's only needed for MS builds.
```

Comment:
I think using a static local achieves this better but since your following the existing pattern then maybe something like that is better as a follow-up PR.

Reply:
AFAIK: we can't do a static-local.  That ends up causing problems with multiple-loads of the AST/loading from serialization/etc.